tests/xharness/Logging/Log.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.IO;
+using System.Text;
+
+namespace xharness.Logging {
+
+	public abstract class Log : TextWriter, ILog {
+		public ILogs Logs { get; private set; }
+		public string Description { get; set; }
+		public bool Timestamp { get; set; } = true;
+
+		public static string XML_LOG = "XML log";
+		public static string NUNIT_RESULT = "NUnit results";
+		public static string SYSTEM_LOG = "System log";
+		public static string COMPANION_SYSTEM_LOG = "System log (companion)";
+		public static string BUILD_LOG = "Build log";
+		public static string TEST_LOG = "Test log";
+		public static string EXTENSION_TEST_LOG = "Extension test log";
+		public static string EXECUTION_LOG = "Execution log";
+
+		protected Log (ILogs logs)
+		{
+			Logs = logs;
+		}
+
+		protected Log (ILogs logs, string description)
+		{
+			Logs = logs;
+			Description = description;
+		}
+
+		public abstract string FullPath { get; }
+
+		public virtual void WriteImpl (string value)
```

Comment:
>WriteImpl [](start = 22, length = 9)

This should rather be protected and not in the interface as it's internal implementation of the class.
You might even make it abstract on this level as well as the `Write` methods and require implementation from the non-abstract implementations of this class

Reply:
100% agree, but it is used by one of the other classes. I need to find a nice way not to do it like this. Will come in another PR. But you are right